pkg/autoscaler/collector.go

```diff
@@ -271,25 +291,40 @@ func (c *collection) currentMetric() *av1alpha1.Metric {
 // record adds a stat to the current collection.
 func (c *collection) record(stat Stat) {
 	// Proxied requests have been counted at the activator. Subtract
-	// AverageProxiedConcurrentRequests to avoid double counting.
-	c.buckets.Record(*stat.Time, stat.PodName, stat.AverageConcurrentRequests-stat.AverageProxiedConcurrentRequests)
+	// them to avoid double counting.
+	c.concurrencyBuckets.Record(*stat.Time, stat.PodName, stat.AverageConcurrentRequests-stat.AverageProxiedConcurrentRequests)
+	c.opsBuckets.Record(*stat.Time, stat.PodName, stat.RequestCount-stat.ProxiedRequestCount)
 }
 
 // stableAndPanicConcurrency calculates both stable and panic concurrency based on the
 // current stats.
 func (c *collection) stableAndPanicConcurrency(now time.Time) (float64, float64, error) {
 	spec := c.currentMetric().Spec
 
-	c.buckets.RemoveOlderThan(now.Add(-spec.StableWindow))
+	return stableAndPanicStats(now, spec.StableWindow, spec.PanicWindow, c.concurrencyBuckets)
+}
+
+// stableAndPanicConcurrency calculates both stable and panic OPS based on the
+// current stats.
+func (c *collection) stableAndPanicOPS(now time.Time) (float64, float64, error) {
+	spec := c.currentMetric().Spec
```

Comment:
Should this stay in `stableAndPanicStats`? Seems like these functions don't need to be concerned with that. Also reduces the signature of `stableAndPanicStats`, which is great as well.

Reply:
Done